validation/ice_ocean_model/ice_ocean_model.jl

```diff
@@ -0,0 +1,344 @@
+        else
+            radiation = filter(flux isa RadiativeEmission, ice.model.external_thermal_fluxes.top) |> first
+        end
+
+        stefan_boltzmann_constant = radiation.stefan_boltzmann_constant
+        reference_temperature = radiation.reference_temperature
+    catch
+    end
+
+    FT = eltype(ocean.model.grid)
+
+    return IceOceanModel(clock,
+                         ocean.model.grid,
+                         ice,
+                         previous_ice_thickness,
+                         previous_ice_concentration,
+                         ocean,
+                         solar_insolation,
+                         convert(FT, ocean_density),
+                         convert(FT, ocean_heat_capacity),
+                         convert(FT, ocean_emissivity),
+                         convert(FT, stefan_boltzmann_constant),
+                         convert(FT, reference_temperature))
+end
+
+time(coupled_model::IceOceanModel) = coupled_model.clock.time
+
+function compute_air_sea_flux!(coupled_model)
+    ocean = coupled_model.ocean
+    ice = coupled_model.ice
+
+    T = ocean.model.tracers.T
+    Nx, Ny, Nz = size(ocean.model.grid)
+
+    grid = ocean.model.grid
+    arch = architecture(grid)
+
+    σ = coupled_model.stefan_boltzmann_constant
+    ρₒ = coupled_model.ocean_density
+    cₒ = coupled_model.ocean_heat_capacity
+    Tᵣ = coupled_model.reference_temperature
+    Qᵀ = T.boundary_conditions.top.condition
+    Tₒ = ocean.model.tracers.T
+    hᵢ = ice.model.ice_thickness
+    ℵᵢ = ice.model.ice_concentration
+    I₀ = coupled_model.solar_insolation
+
+    launch!(arch, grid, :xy, _compute_air_sea_flux!,
+            Qᵀ, grid, Tₒ, hᵢ, ℵᵢ, I₀,
+            σ, ρₒ, cₒ, Tᵣ)
+end
+
+@kernel function _compute_air_sea_flux!(temperature_flux,
+                                        grid,
+                                        ocean_temperature,
+                                        ice_thickness,
+                                        ice_concentration,
+                                        solar_insolation,
+                                        σ, ρₒ, cₒ, Tᵣ)
+
+    i, j = @index(Global, NTuple)
+
+    Nz = size(grid, 3)
+
+    @inbounds begin
+        T₀ = ocean_temperature[i, j, Nz] # at the surface
+        h = ice_thickness[i, j, 1]
+        ℵ = ice_concentration[i, j, 1]
+        I₀ = solar_insolation[i, j, 1]
+    end
+
+    # Radiation model
+    ϵ = 1 # ocean emissivity
+    ΣQᵀ = ϵ * σ * (T₀ + Tᵣ)^4 / (ρₒ * cₒ)
+
+    # Also add solar insolation
+    ΣQᵀ += I₀ / (ρₒ * cₒ)
+
+    # Set the surface flux only if ice-free
+    Qᵀ = temperature_flux
+
+    # @inbounds Qᵀ[i, j, 1] = 0 #(1 - ℵ) * ΣQᵀ
+end
+
+function time_step!(coupled_model::IceOceanModel, Δt; callbacks=nothing)
+    ocean = coupled_model.ocean
+    ice = coupled_model.ice
+    liquidus = ice.model.phase_transitions.liquidus
+    grid = ocean.model.grid
+    ice.Δt = Δt
+    ocean.Δt = Δt
+
+    fill_halo_regions!(h)
```

Comment:
where is this `h` coming from?

Reply:
Hm, there should be an

```julia
h = ice.model.ice_thickness
```

before.

it's taking it from the global scope here so it just happened to work.

Reply:
fixed here https://github.com/CliMA/ClimaSeaIce.jl/pull/20